jwst/associations/lib/rules_level3.py

```diff
@@ -237,6 +245,53 @@ def _init_hook(self, item):
         super(Asn_Lv3Image, self)._init_hook(item)
 
 
+@RegistryMarker.rule
+class Asn_Lv3ImageBackground(AsnMixin_AuxData, AsnMixin_Science):
+    """Level 3 Background Image Association
+
+    Characteristics:
+        - Association type: ``image3``
+        - Pipeline: ``calwebb_image3``
+        - Non-TSO
+        - Non-WFS&C
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        # Setup constraints
+        self.constraints = Constraint([
+            Constraint_Optical_Path(),
+            Constraint_Target(association=self),
+            Constraint_Image(),
+            DMSAttrConstraint(
+                name='wfsvisit',
+                sources=['visitype'],
+                value='((?!wfsc).)*',
+                required=False
```

Comment:
Does the `required=False` turn this into a constraint that excludes WFS&C exposures from this type of ASN? If so, it's not obvious (at least to me) from the naming/syntax. To me saying that something is not required is equivalent to saying it's optional and hence could be included, but doesn't need to be.

Reply:
I'll wait with you to hear from Jonathan on the reality of this, but my understanding is that the required check is to prevent member candidates without an entry from causing problems. The constraint on value will exclude the WFS&C entries because they should always contain a value for `wfsvisit`.

Reply:
Oh yeah, I didn't notice (till now) the presence of the "!" in front of "wfsc", hence it is excluding wfsc exposures. Never mind!

Reply:
FYI: The `required` option, which yes there needs to be documentation on this, refers to whether the sources are actually available.

Using the current rule as an example: If the column `visitype` was not present in the pool or the value is a non-value, such as `null`, the constraint could not be checked. The default, `required=True`, says the column has to exist, and the constraint would fail. However, with `required=False`, the constraint assumes truth.

Two reasons for this, the original one being that different incantations of pools generated by SDP rapidly changed what columns were in them during development. This allowed using older pools.

The second reason is that what columns are filled in is mode-dependent. If a column is not relevant for a particular mode, a "non-value" is assigned. Depending on the original source of each column, the non-value could be any number of things, such as `null`, `None`, etc. This option allows a constraint to be defined that may cover a number of different modes and ignore sometimes-defined columns amongst those modes. This happens most often when a constraint covers something like "all imaging exposure types among all instruments", but a related constraint may be applied to a column that is only defined for a particular instrument in imaging mode.